### PR TITLE
dcs: init at unstable-2021-04-07

### PR DESCRIPTION
**Pull request by cole-h**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120007` at abe917d0 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>dcs</li>
    <li>perl530Packages.CSSMinifier</li>
    <li>perl532Packages.CSSMinifier</li>
  </ul>
</details>

File: pkgs/tools/text/dcs/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, python3
+, perl
+, zopfli
+}:
+buildGoModule {
```
Comment (SuperSandro2000) [inline pkgs/tools/text/dcs/default.nix:8]:
```suggestion
}:

buildGoModule {
```

File: pkgs/tools/text/dcs/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  vendorSha256 = "l2mziuisx0HzuP88rS5M+Wha6lu8P036wJYZlmzjWfs=";
+
+  # Depends on dcs binaries
+  doCheck = false;
+
+  nativeBuildInputs = [
+    python3.pkgs.slimit
+    perl
+    perl.pkgs.CSSMinifier
```
Comment (SuperSandro2000) [inline pkgs/tools/text/dcs/default.nix:27]:
```suggestion
    perl.withPackages (p: [ p.CSSMinifier ])
```

File: pkgs/tools/text/dcs/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    sha256 = "N+6BXlKn1YTlh0ZdPNWa0nuJNcQtlUIc9TocM8cbzQk=";
+  };
+
+  vendorSha256 = "l2mziuisx0HzuP88rS5M+Wha6lu8P036wJYZlmzjWfs=";
+
+  # Depends on dcs binaries
+  doCheck = false;
+
+  nativeBuildInputs = [
+    python3.pkgs.slimit
```
Comment (SuperSandro2000) [inline pkgs/tools/text/dcs/default.nix:25]:
```suggestion
    python3Packages.slimit
```

File: pkgs/tools/text/dcs/default.nix [hunk 4]
```diff
@@ -0,0 +1,47 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, python3
```
Comment (SuperSandro2000) [inline pkgs/tools/text/dcs/default.nix:4]:
```suggestion
, python3Packages
```

File: pkgs/top-level/perl-packages.nix [hunk 5]
```diff
@@ -4481,6 +4481,19 @@ let
...
+  CSSMinifier = buildPerlPackage {
+    pname = "CSS-Minifier";
+    version = "0.01";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/P/PM/PMICHAUX/CSS-Minifier-0.01.tar.gz";
+      sha256 = "0Kk0m46LfoOrcM+IVM+7Qv8pwfbHyCmPIlfdIaoMf+8=";
+    };
+    meta = {
+      description = "Perl extension for minifying CSS";
+      license = lib.licenses.artistic1;
```
Comment (SuperSandro2000) [inline pkgs/top-level/perl-packages.nix:4493]: Missing maintainer

**Comment by SuperSandro2000** [comment]
Any idea why it is broken on aarch64?

**Review by stigtsp — DISMISSED** [review]
Pls change commit message from `perl.pkgs.CSSMinifier: init at 0.01` to `perlPackages.CSSMinifier: init at 0.01`

File: pkgs/top-level/perl-packages.nix (hunk 5 above)
Comment (cole-h) [reply pkgs/top-level/perl-packages.nix:4493]: It looks like some / most packages in here don't have maintainers, so I figured that was the norm -- happy to add us if that's not the case, however.

File: pkgs/tools/text/dcs/default.nix (hunk 3 above)
Comment (cole-h) [reply pkgs/tools/text/dcs/default.nix:25]: Any reason in particular? They're the same, AFAICS.

**Comment by cole-h** [comment]
Brokenness on aarch64 might be related to https://github.com/Debian/dcs/blob/da46accc4d55e9bfde1a6852ac5a9e730fcbbb2c/index/myvarint.go#L115 -- `nix-build -A pkgsCross.aarch64-multiplatform.dcs` failed with:

```
index/merge.go:428:22: undefined: myPostingLast
index/read.go:289:9: undefined: myPostingList
index/read.go:299:9: undefined: myPostingAnd
index/read.go:309:9: undefined: myPostingOr
```

File: pkgs/tools/text/dcs/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/tools/text/dcs/default.nix:26]: I think python3Packages might be easier to overwrite but the difference is minimal.

Comment (cole-h) [reply pkgs/tools/text/dcs/default.nix]: That's reasonable enough.

File: pkgs/tools/text/dcs/default.nix [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  doCheck = false;
+
+  nativeBuildInputs = [
+    python3Packages.slimit
+    (perl.withPackages (p: [ p.CSSMinifier ]))
+    zopfli
+  ];
+
+  postBuild = ''
+    make -C static
```
Comment (SuperSandro2000) [inline pkgs/tools/text/dcs/default.nix:32]:
```suggestion
    make -C static -j$NIX_BUILD_CORES
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 120007` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>dcs</li>
<li>perl530Packages.CSSMinifier</li>
<li>perl532Packages.CSSMinifier</li>
</ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**